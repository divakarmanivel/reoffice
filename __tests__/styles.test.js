import { applyStyles, applyImageProps, styleSerializer } from '../src/styles/styles';

it('children can be colored with a hex', () => {
  const props = {
    color: '#ffffff',
    backgroundColor: '#ff0000',
  };

  expect(applyStyles(props)).toMatchSnapshot();
});

it('children can be styled', () => {
  const props = {
    color: '#ffffff',
    backgroundColor: '#ff0000',
    fontSize: '40',
    fontFace: 'Tahoma',
    bold: true,
    border: true,
    underline: true,
    italic: true,
    highlight: '#ff2222',
    borderSize: '40',
    borderColor: '#ff3',
    link: 'www.google.com',
  };

  expect(applyStyles(props)).toMatchSnapshot();
});

it('image size can be adjusted with style prop', () => {
  const props = {
    style: {
      width: '200',
      height: '200',
    },
  };

  expect(applyImageProps(props)).toMatchSnapshot();
});

it('image size can be adjusted with width and height prop', () => {
  const props = {
    width: '200',
    height: '200',
  };

  expect(applyImageProps(props)).toMatchSnapshot();
});

it('table styles can be serialized according to open office xml layout', () => {
  const styles = {
    bold: true,
    size: '20',
    color: 'mistyrose',
    align: 'center',
    vAlign: 'center',
    fontFamily: 'Tahoma',
    fill: '#ffffff',
    cellColWidth: 4261,
  };

  expect(styleSerializer(styles)).toMatchSnapshot();
});
