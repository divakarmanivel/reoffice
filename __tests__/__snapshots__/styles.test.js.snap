// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`children can be colored with a hex 1`] = `
{
  "back": "ff0000",
  "bold": undefined,
  "border": undefined,
  "borderColor": false,
  "borderSize": undefined,
  "color": "ffffff",
  "font_face": undefined,
  "font_size": undefined,
  "highlight": false,
  "italic": undefined,
  "link": undefined,
  "underline": undefined,
}
`;

exports[`children can be styled 1`] = `
{
  "back": "ff0000",
  "bold": true,
  "border": true,
  "borderColor": "ffff33",
  "borderSize": "40",
  "color": "ffffff",
  "font_face": "Tahoma",
  "font_size": "40",
  "highlight": "ff2222",
  "italic": true,
  "link": "www.google.com",
  "underline": true,
}
`;

exports[`image size can be adjusted with style prop 1`] = `
{
  "cx": 200,
  "cy": 200,
}
`;

exports[`image size can be adjusted with width and height prop 1`] = `
{
  "cx": 200,
  "cy": 200,
}
`;

exports[`table styles can be serialized according to open office xml layout 1`] = `
{
  "align": "center",
  "b": true,
  "cellColWidth": 4261,
  "color": "ffe4e1",
  "fill": "ffffff",
  "fontFamily": "Tahoma",
  "sz": "20",
  "vAlign": "center",
}
`;
